pennylane/beta/queuing/measure.py

```diff
@@ -17,16 +17,157 @@
+        self.data = []
+
+        # Queue the measurement process
+        self.queue()
+
+    @property
+    def wires(self):
+        r"""The wires the measurement process acts on."""
+        if self.obs is not None:
+            return self.obs.wires
+        return self._wires
+
+    @property
+    def eigvals(self):
+        r"""Eigenvalues associated with the measurement process.
+
+        If the measurement process has an associated observable,
+        the eigenvalues will correspond to this observable. Otherwise,
+        they will be the eigenvalues provided when the measurement
+        process was instantiated.
+
+        Note that the eigenvalues are not guaranteed to be in any
+        particular order.
+
+        **Example:**
+
+        >>> m = MeasurementProcess(Expectation, obs=qml.PauliX(wires=1))
+        >>> U.eigvals
+        >>> array([1, -1])
+
+        Returns:
+            array: eigvals representation
+        """
+        if self.obs is not None:
+            try:
+                return self.obs.eigvals
+            except NotImplementedError:
+                pass
+
+        return self._eigvals
+
+    def expand(self):
+        """Expand the measurement of an observable to a unitary
+        rotation and a measurement in the computational basis.
+
+        Returns:
+            .QuantumTape: a quantum tape containing the operations
+            required to diagonalize the observable
+
+        **Example**
+
+        Consider a measurement process consisting of the expectation
+        value of an Hermitian observable:
+
+        >>> H = np.array([[1, 2], [2, 4]])
+        >>> obs = qml.Hermitian(H, wires=['a'])
+        >>> m = MeasurementProcess(Expectation, obs=obs)
+
+        Expanding this out:
+
+        >>> tape = m.expand()
+
+        We can see that the resulting tape has the qubit unitary applied,
+        and a measurement process with no observable, but the eigenvalues
+        specified:
+
+        >>> print(tape.operations)
+        [QubitUnitary(array([[-0.89442719,  0.4472136 ],
+              [ 0.4472136 ,  0.89442719]]), wires=['a'])]
+        >>> print(tape.measurements[0].eigvals)
+        [0. 5.]
+        >>> print(tape.measurements[0].obs)
+        None
+        """
+        if self.obs is None:
+            raise NotImplementedError("Cannot expand a measurement process with no observable.")
+
+        from pennylane.beta.tapes import QuantumTape  # pylint: disable=import-outside-toplevel
+
+        with QuantumTape() as tape:
+            self.obs.diagonalizing_gates()
+            MeasurementProcess(self.return_type, wires=self.obs.wires, eigvals=self.obs.eigvals)
+
+        return tape
+
+    def queue(self):
+        """Append the measurement process to an annotated queue."""
+        if self.obs is not None:
+            QueuingContext.update_info(self.obs, owner=self)
+            QueuingContext.append(self, owns=self.obs)
+        else:
+            QueuingContext.append(self)
```

Comment:
Is the ``if`` for ``expval`` etc, and the ``else`` for ``probs``?

Reply:
yep!